packages/perseus-editor/src/components/graph-settings.tsx

```diff
@@ -101,13 +111,13 @@ const GraphSettings = createReactClass({
         ) {
             this.setState(this.stateFromProps(nextProps));
         }
-    },
+    }
 
-    componentWillUnmount: function () {
+    componentWillUnmount() {
         this._isMounted = false;
-    },
+    }
 
-    stateFromProps: function (props) {
+    stateFromProps(props) {
```

Comment:
Can you mark this as `static`?  It will mean a few changes to call sites, but it'll ensure we don't "cheat" in this function ever and start accessing instance fields.